test/e2e/nodeportlocal_test.go

```diff
@@ -125,6 +125,13 @@ func getNPLAnnotation(t *testing.T, data *TestData, r *require.Assertions, testP
 			ann := pod.GetAnnotations()
 			t.Logf("Got annotations %v for Pod with IP %v", ann, testPodIP.ipv4.String())
 			nplAnn, found = ann[types.NPLAnnotationKey]
+			if conditionFn != nil {
+				var nplAnnotation types.NPLAnnotation
+				json.Unmarshal([]byte(nplAnn), &nplAnnotation)
```

Comment:
Doesn't `nplAnn` contain multiple NPLAnnotations?
I think you can merge `getNPLAnnotation` and `getNPLAnnotations` to reduce redundancy:
```
func getNPLAnnotations(t *testing.T, data *TestData, r *require.Assertions, testPodName string, conditionFn func(types.NPLAnnotation) bool) ([]types.NPLAnnotation, string) {
	var nplAnnotations []types.NPLAnnotation
	var testPodIP *PodIPs

	var err error
	maxRetries := 0
	podTimeout := defaultTimeout
	if testOptions.enableAntreaIPAM {
		// If enableAntreaIPAM and agent restarted, we'll get error "http2: client connection lost" when getting Pod annotations.
		// Add more retries and reduce the timeout to handle this case.
		maxRetries = 4
		podTimeout = 18 * time.Second
	}
	for i := 0; i <= maxRetries; i++ {
		_, err = data.PodWaitFor(podTimeout, testPodName, data.testNamespace, func(pod *corev1.Pod) (bool, error) {
			var err error
			if pod.Status.Phase != corev1.PodRunning {
				return false, nil
			}

			podIPStrings := sets.New[string](pod.Status.PodIP)
			for _, podIP := range pod.Status.PodIPs {
				ipStr := strings.TrimSpace(podIP.IP)
				if ipStr != "" {
					podIPStrings.Insert(ipStr)
				}
			}

			testPodIP, err = parsePodIPs(podIPStrings)
			if err != nil || testPodIP.ipv4 == nil {
				return false, nil
			}

			ann := pod.GetAnnotations()
			t.Logf("Got annotations %v for Pod with IP %v", ann, testPodIP.ipv4.String())
			nplAnn, found := ann[types.NPLAnnotationKey]
			if !found {
				return false, nil
			}
			json.Unmarshal([]byte(nplAnn), &nplAnnotations)
			if conditionFn != nil {
				for _, annotation := range nplAnnotations {
					if !conditionFn(annotation) {
						return false, nil
					}
				}
			}
			return found, nil
		})
		if err == nil {
			break
		}
		t.Logf("Got error when get Pod annotations, err=%+v", err)
		time.Sleep(time.Millisecond * 100)
	}
	r.NoError(err, "Poll for Pod check failed")
	return nplAnnotations, testPodIP.ipv4.String()
}
```

Reply:
Updated, thanks.